anta/input_models/snmp.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2023-2025 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+"""Module containing input models for SNMP tests."""
+
+from __future__ import annotations
+
+from pydantic import BaseModel, ConfigDict
+
+from anta.custom_types import HashingAlgorithm, SnmpEncryptionAlgorithm, SnmpVersion
+
+
+class SnmpUser(BaseModel):
+    """Model for a SNMP User."""
+
+    model_config = ConfigDict(extra="forbid")
+    username: str
+    """SNMP user name."""
+    group_name: str | None = None
+    """SNMP group for the user. Required field in the `VerifySnmpUser` test."""
+    version: SnmpVersion | None = None
+    """SNMP protocol version. Required field in the `VerifySnmpUser` test."""
```

Comment:
This model is being used by `VerifySnmpUser` only and these fields are required in it so they can never be `None`. You could remove the validator in the test.
```suggestion
    group_name: str
    """SNMP group for the user."""
    version: SnmpVersion
    """SNMP protocol version."""
```

Reply:
Updated. Thanks!!